addons/account/models/company.py

```diff
@@ -642,57 +644,51 @@ def _check_hash_integrity(self):
             # We need the `sudo()` to ensure that all the moves are searched, no matter the user's access rights.
             # This is required in order to generate consistent hashes.
             # It is not an issue, since the data is only used to compute a hash and not to return the actual values.
-            field_names = [
-                *self.env['account.move']._get_integrity_hash_fields(),
-                'inalterable_hash',
-                'journal_id',
-                'move_type',
-                'sequence_prefix',
-                'sequence_number',
-                'secure_sequence_number',
-            ]
-            moves = self.env['account.move'].sudo().search_fetch(
+            query = self.env['account.move'].sudo()._search(
                 domain=[
                     ('journal_id', '=', journal.id),
                     ('inalterable_hash', '!=', False),
                 ],
-                field_names=field_names,
-                order="secure_sequence_number ASC NULLS LAST, sequence_prefix, sequence_number ASC"
+                order="secure_sequence_number ASC NULLS LAST, sequence_prefix, sequence_number ASC",
             )
-
-            if not moves:
-                results.append({
-                    'journal_name': journal.name,
-                    'restricted_by_hash_table': 'V',
-                    'status': 'no_data',
-                    'msg_cover': _('There is no journal entry flagged for accounting data inalterability yet.'),
-                })
-                continue
-
             prefix2result = defaultdict(lambda: {
                 'first_move': self.env['account.move'],
                 'last_move': self.env['account.move'],
                 'corrupted_move': self.env['account.move'],
             })
             last_move = self.env['account.move']
-            current_hash_version = 1
-            for move in moves:
-                prefix_result = prefix2result[move.sequence_prefix]
-                if prefix_result['corrupted_move']:
+            self.env.execute_query(SQL("DECLARE hashed_moves CURSOR FOR %s", query.select()))
+            while move_ids := self.env.execute_query(SQL("FETCH %s FROM hashed_moves", INTEGRITY_HASH_BATCH_SIZE)):
+                moves = self.env['account.move'].browse(move_id[0] for move_id in move_ids)
```

Comment:
Otherwise the cache will keep growing after each batch, defeating the purpose of the PR
```suggestion
                self.env.invalidate_all()
                moves = self.env['account.move'].browse(move_id[0] for move_id in move_ids)
```

Reply:
Why invalidating the whole environment ? Shouldn't `invalidate_model` or `invalidate_recordset` be enough for removing the 1000 moves from the cache ?

Reply:
Okkk, let's take no risk then